anna-sala/pre-curso/airlines-pro.js

```diff
@@ -0,0 +1,258 @@
+    { id: 08, to: 'Shangai', from: 'Barcelona', cost: 800, scale: true },
+    { id: 09, to: 'Sydney', from: 'Barcelona', cost: 150, scale: true },
+    { id: 10, to: 'Tel-Aviv', from: 'Madrid', cost: 150, scale: false } ];
+let userName;
+userName = prompt('¿Como te llamas?');
+alert('Bienvenido a Skylab Airlines, ' + userName);
+
+airlines();
+
+function airlines(){
+    seeAllFlights();
+    averageCost();
+    seeScales();
+    afternoonFlights();
+}
+
+
+function seeAllFlights(){
+    for (let i=0; i<flights.length; i++){
+            if ( flights[i].scale){
+            console.log("ID: "+ flights[i].id + ". El vuelo con origen " + flights[i].from + " y destino " + flights[i].to + " tiene un coste de " + flights[i].cost + "€ y si realiza escalas")
+            }
+            else {
+            console.log("ID: "+ flights[i].id + ". El vuelo con origen " + flights[i].from + " y destino " + flights[i].to + " tiene un coste de " + flights[i].cost + "€ y no realiza escalas")
+            }
+    }
+}
+
+function averageCost (){
+    let total=0;
+    let average;
+    for (let i=0; i<flights.length; i++){
+        total +=flights[i].cost; 
+    }
+average = total/flights.length;
+console.log("El precio medio de los vuelos es " + average.toFixed(2) + "€");
+}
+
+function seeScales (){
+    console.log("Los vuelos que tienen escalas son: ");
+    for (let i=0; i<flights.length; i++){
+        if (flights[i].scale===true){
+        console.log(" El vuelo con identificador: " + flights[i].id + " con origen " + flights[i].from + " y destino " + flights[i].to + "y tiene un coste de " + flights[i].cost);
+        }
+}
+}
+
+function afternoonFlights(){
+    console.log("Los destinos de los últimos velos del día son:  ")
+    for (let i=6; i<flights.length; i++){
+            console.log(" El vuelo con identificador: " + flights[i].id + " y destino " + flights[i].to);
+    }
+}
+
+chooseAdminUser();
+function chooseAdminUser(){
+    let adminUser;
+    adminUser = prompt(userName +", eres Admin o Usuario?, por favor escribe admin o usuario.");
+        if (adminUser==="admin" || adminUser==="Admin"){
+            adminFunctions();
+        }
+        else if (adminUser==="" || adminUser==null){
+            alert ("Por favor escriba solo 'admin' o 'usuario'.")
+            chooseAdminUser();
+        }
+        else {
+            userFunctions();
+        }
+}
+
+
+function adminFunctions(){
+    let adminOptions = prompt ("Quieres crear un nuevo vuelo o elimar existentes? Escribe 'nuevo' o 'eliminar'. Presiona Cancelar para salir. ");
+        if (adminOptions ==="nuevo" || adminOptions ==="Nuevo" ){
+            createNewFlightAdmin();
+        }
+        else if (adminOptions===null || adminOptions===""){
+            console.log("Gracias por volar con Skylab Airlines, hasta pronto! ");
+            outOfAdmin();
+        }
+        else {
+            eliminateFlightsAdmin();
+        }
+
+            function createNewFlightAdmin(){
```

Comment:
Do not declare functions inside other functions. Declare it out and call it inside then.